arangod/Aql/IResearchViewExecutor.cpp

```diff
@@ -1376,11 +1422,31 @@ size_t IResearchViewMergeExecutor<ordered, materializeType>::skipAll() {
   TRI_ASSERT(this->_filter != nullptr);
 
   size_t skipped = 0;
-
-  while (_heap_it.next()) {
-    skipped++;
+  if (_heap_it.size()) {
+    for (auto& segment : _segments) {
+      TRI_ASSERT(segment.docs);
+      if (this->infos().filterConditionIsEmpty()) {
+        TRI_ASSERT(segment.segmentIndex < this->_reader->size());
+        auto const live_docs_count =  (*this->_reader)[segment.segmentIndex].live_docs_count();
+        TRI_ASSERT(segment.segmentPos <= live_docs_count);
+        skipped += live_docs_count - segment.segmentPos;
+        segment.segmentPos = live_docs_count;
+      } else {
+       skipped +=  calculateSkipAllCount(this->infos().countApproximate(),
+                                         segment.segmentPos, segment.docs.get());
+      }
```

Comment:
should `segment.segmentPos` be set to the size of the segment here?

Reply:
To be fully consistent - it should be increased by "skipped from segment" count. Will fix.